relayer/src/chain/cosmos.rs

```diff
@@ -1364,12 +1346,11 @@ fn update_client_from_tx_search_response(
             IbcEvent::UpdateClient(update) => Some(update),
             _ => None,
         })
-        .filter(|update| {
+        .find(|update| {
```

Comment:
I am not that this has the same semantics as the original code in case there is more than one event at this stage.

Initially, we were adding the events to a Vec and then returning the last one with `pop`, which we then changed to a call to `last` on the iterator, but now with `find`, the first element that matches the filter will be returned instead of the last.

Reply:
Yes it's true. I changed it because, currently, first or last are both ok. 
Here we are looking at multiple messages inside a single Tx. First one will be for the message that actually installed the consensus state. Subsequent client update messages for same height are noop-s wrt to chain state. For the same reason we get the first Tx. So this is just saying, from that first Tx (`response` here), get the first matching client update message event.